apiserver/internal/handlers/resources/resources.go

```diff
@@ -167,8 +167,6 @@ func (h *ResourceHandler) upload(service ResourceService, req *http.Request, use
 			RetrievedByType: coreresource.User,
 			Size:            uploaded.size,
 			Fingerprint:     uploaded.fingerprint,
-			Origin:          charmresource.OriginUpload,
-			Revision:        -1,
```

Comment:
The point about checking the origin and revision above is not a problem here due to @hmlanigan's changes https://github.com/juju/juju/pull/18946